Sources/PackageLoading/Target+PkgConfig.swift

```diff
@@ -252,17 +253,26 @@ public func removeDefaultFlags(cFlags: [String], libs: [String]) throws -> ([Str
     )
 }
 
-public struct PkgConfigDiagnosticLocation: DiagnosticLocation {
-    public let pcFile: String
-    public let target: String
+extension ObservabilityMetadata {
+    public static func pkgConfig(pcFile: String, targetName: String) -> Self {
+        var metadata = ObservabilityMetadata()
+        metadata.pcFile = "\(pcFile).pc"
+        metadata.targetName = targetName
+        return metadata
+    }
+}
 
-    public init(pcFile: String, target: String) {
-        self.pcFile = pcFile
-        self.target = target
+extension ObservabilityMetadata {
+    public var pcFile: String? {
```

Comment:
Nice.

Reply:
with the current architecture the clients would need to know what to look for in the metadata, and there is no tie in back to the scope. these extensions however are usually internal such that only if you import the right module you can access them. happy to do something better if you have an idea in mind.